render_bundle/lib/charm_bundle_generator.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Generates any charm bundle from user provided Application and Relation objects."""
+
+from pathlib import Path
+from typing import Optional
+
+from jinja2 import Template
+from pydantic import BaseModel
+
+
+def write_to_file(file: str, content: str):
```

Comment:
```suggestion
def _write_to_file(file: str, content: str):
```
I would probably make this private, I don't think this should be part of the public API.

Reply:
Done